flowcraft/flowcraft.py

```diff
@@ -286,6 +295,58 @@ def copy_project(path):
                 join(target_dir, "profiles.config"))
 
 
+def add_cache_dir(path):
+    """
+    Function that checks if SINGULARITY_CACHEDIR or NXF_SINGULARITY_CACHEDIR
+    are defined and if so writes the path to the custom.config overriding the
+    default flowcraft singularity.cacheDir defined in nextflow.config
+
+    Parameters
+    ----------
+    path: str
+        path to the directory where the file will be written
+
+    """
+
+    # Get target directory
+    target_dir = dirname(path)
+
+    env_var = False
+    # Checks whether the environmental variables are defined or not. If they
+    # aren't then raise a warning and use the default.
+    try:
+        env_var = os.environ.get("SINGULARITY_CACHEDIR")
+    except KeyError:
+        try:
+            env_var = os.environ.get("NXF_SINGULARITY_CACHEDIR")
+        except KeyError:
+            logger.error(colored_print("No environmental variables set with "
+                                       "the names: SINGULARITY_CACHEDIR or "
+                                       "NXF_SINGULARITY_CACHEDIR. Default cache"
+                                       " dir will be used instead (see "
+                                       "nextflow.config for more details)",
+                                       "red_bold"))
+
+    # if env_var is different from False then read the user.config and write to
+    # the new singularity.cacheDir fetched from the os.environ respective
+    # key
+    if env_var:
+        with open(os.path.join(target_dir, "custom.config"), "r+") as \
```

Comment:
Why are you creating a new configuration file just to store the singularity cache dir? Why not use container.config file (suggested by @cimendes)?

Reply:
Not sure about this. When I first though of `containers.config` I thought of dockerhub and so on, not on env variables but maybe this is better discussed in an issue. @ODiogoSilva was also somehow skeptical that this was the best solution for the problem. Then, maybe the would be to open an issue for this.